resource_retriever/include/resource_retriever/retriever.h

```diff
@@ -29,65 +29,15 @@
 #ifndef RESOURCE_RETRIEVER__RETRIEVER_H_
 #define RESOURCE_RETRIEVER__RETRIEVER_H_
 
-#include <cstdint>
-#include <memory>
-#include <stdexcept>
-#include <string>
+#ifdef _MSC_VER
+#pragma message("Including header <resource_retriever/retriever.h> is deprecated,")
+#pragma message("include <resource_retriever/retriever.hpp> instead.")
+#else
+// *INDENT-OFF* (prevent uncrustify from adding indention below)
+#warning Including header <resource_retriever/retriever.h> is deprecated, \
+include <resource_retriever/retriever.hpp> instead.
+#endif
```

Comment:
Hm, do we need the separate `pragma` for Windows?  It doesn't seem like we are doing that [elsewhere](https://github.com/ros/pluginlib/blob/ros2/pluginlib/include/pluginlib/class_loader.h), but that is actually the only example I can find in our current codebase.

Reply:
I mostly blindly copied it from `class_loader` https://github.com/ros/class_loader/blob/8864be372a83921f15906ae80752be9d6a300b81/include/class_loader/class_loader_core.h#L35-L42

This Stack Overflow answer suggests the answer is yes https://stackoverflow.com/questions/171435/portability-of-warning-preprocessor-directive - maybe no code on windows has tried to include `<pluginlib/class_loader.h>`?

Reply:
> This Stack Overflow answer suggests the answer is yes https://stackoverflow.com/questions/171435/portability-of-warning-preprocessor-directive - maybe no code on windows has tried to include `<pluginlib/class_loader.h>`?

Ah, interesting.  OK, totally fine with me, just missed where this came from.